crates/spk-cli/common/src/flags.rs

```diff
@@ -980,6 +981,21 @@ impl Repositories {
             }
             repos.push((name.into(), repo.into()));
         }
+        let has_origin = repos.iter().find(|(n, _)| n == "origin").is_some();
+        if !disabled.contains("origin") && !has_origin {
+            // the origin repo is considered a default and specifically
+            // added at the very end of any command line flags if needed
+            match storage::remote_repository::<_, NormalizedTagStrategy>("origin").await {
+                Err(spk_storage::Error::SPFS(spfs::Error::UnknownRemoteName(_))) => {}
+                Err(err) => return Err(err.into()),
+                Ok(mut origin) => {
+                    if self.legacy_spk_version_tags {
+                        origin.set_legacy_spk_version_tags(true);
```

Comment:
The pin at time section is missing here. I suggest refactoring the loop body above to reuse the same logic.

Reply:
Our default origin never had an associated pin time...

Reply:
The old code would respect the pin option if given, no?

Reply:
Yes, and for user provided values it still does, but the injected origin repo was always `("origin", None)` so handling the pin in this portion of the code would be redundant

Reply:
Okay, I see what you mean. IMO it would still be an improvement to unify the code but I'll approve pending CI passing.

Reply:
Ya, I agree I'll see if there's a trivial way to work a closure in or something

Reply:
I gave this a try but the use of async makes it difficult to use a closure or similar reuse pattern. Happy to have someone iterate though...

Reply:
The use of `Option::or` on line 976 means the "origin" repo would still respect the `--when` command line flag, and it is reasonable to assume that `--when` would apply to the default origin repo.

I have a PR on the way to restore this behavior.